### ISSUE TITLE
Performance issue about select with large dataset

### ISSUE THREAD

**Issue by codTango** [ant-design/ant-design#5854]
### What problem does this feature solve?
I'm using antd.select in my project, but the performance is not ideal when using with a very large amount of data. For example, loading 10000 items to the select, the initial render time will be slow and opening and scrolling dropdown menu is super laggy in this case, however, this is just the normal data size in my project, it would be even larger than that.

### What does the proposed API look like?
infinite scroll or virtualized list in the dropdown menu would be perfect to solve this issue. Or at least expose an onScroll API, so I can detect the scroll direction and position to dynamically load my data, then allow loading the dropdown menu with the new data on fly.

**Comment by kossel** [comment]
Why would you load 10000 items into a select? how you would find an item from a list of 10000?
also I think you should create the issue with   http://new-issue.ant.design

**Comment by codTango** [comment]
<img width="1128" alt="screen shot 2017-04-21 at 9 52 02 pm" src="https://cloud.githubusercontent.com/assets/5359697/25300422/fafd7712-26dc-11e7-8d96-3e8b8e1c750c.png">
As shown above, I built HOC based on antd's treeSelect and select components, it's a cascading filter and has search feature.

I say 10000 items which is the data set I'm using for test. In the real world, it could be from couple to couple thousands items, or even more than 10000 items per the customer. 

However, if it's only 2000 items for each selector, it's not ideal to load them all at once. Also considering the selector will  be populated by previous selection on fly, that cannot be preloaded. I'm seeking the infinite scroll or virtualized list for the dropdown menu here. 

Also I used that link to create this issue, I selected feature request...
